classes/Gems/Acl/Privilege.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Gems\Acl;
+
+use Gems\Exception\AuthenticationException;
+
+class Privilege
+{
+    public const SEPARATOR = '.';
+
+    public function __construct(
+        public readonly string $name,
+        public readonly ?string $label = null,
+        public readonly ?array $methods = null,
+    )
+    {}
+
+    public function getName(?string $method = null): string
+    {
+        if ($method !== null && $this->methods !== null) {
+            if (!in_array($method, $this->methods)) {
+                throw new AuthenticationException('Method not allowed');
+            }
+            return $this->name . self::SEPARATOR . $method;
```

Comment:
voor de `DELETE` method krijg je dus `name.DELETE`, terwijl je mogelijk op dit moment ook al permissies hebt die op `.delete` eindigen. Is dat een probleem?